Trigger upstream CI on cron schedule (by default) (#4729)
pydata/xarray: .github/PULL_REQUEST_TEMPLATE.md

- [ ] Passes `isort . && black . && mypy . && flake8`
- [ ] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [ ] New functions/methods are listed in `api.rst`

> <sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior by adding a `[test-upstream]` tag in the git commit’s title.</sub>

Reviewer: not sure if "commit title" is a well-known term. Also, it would probably help to state that this has to be done for every commit the CI is supposed to run on.

Maybe something like this instead?
```suggestion
<sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior per commit by adding a `[test-upstream]` tag in the first line of the commit message.</sub>
```

> <sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior per commit by adding a `[test-upstream]` tag in the first line of the commit message.</sub>

Reviewer: ```suggestion
<sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior per commit by adding a `[test-upstream]` tag to the first line of the commit message.</sub>
```

> <sub>By default, the upstream dev CI is disabled on pull request and push events. You can override this behavior by adding a `[test-upstream]` tag in the body of the git commit or the commit’s title.</sub>

Author: > now that it works, we should probably document [test-upstream] somewhere. I think this would fit in the PR template (as a comment?) or the contributing guide.

Is this good enough?

Reviewer: > in the body of the git commit

I don't think we support that. Actually, I actively tried to disable that because when squash merging the PR title and the commit messages of the squashed commits are concatenated, which would mean that if any of the commit messages had that tag the merge commit would have it, too.

That also means that unless the one pressing the merge button modifies it, adding a tag to the PR title will trigger the CI (not sure if we should encourage using this, though).

> Is this good enough or should we add something in the contributing guide too?

I would say this is enough, but I'm not sure. Thoughts, @max-sixty?
